### ISSUE TITLE
Rework Badge format

### ISSUE THREAD

**Issue by jsoref** [argoproj/argo-cd#5260]
If you are trying to resolve an environment-specific issue or have a one-off question about the edge case that does not require a feature then please consider asking a question in argocd slack [channel](https://argoproj.github.io/community/join-slack).

Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

There isn't enough space between the color transition and the left edge of the text to the right of it.

**To Reproduce**

1. Visit https://cd.apps.argoproj.io/api/badge?project=default&revision=true

(Reachable from https://cd.apps.argoproj.io/settings/projects/default )

**Expected behavior**

There should be one more space between the color transition and the beginning of the text (`OutOfSync`) -- to match the ~2 spaces between the right edge of `Degraded` and the transition.

**Screenshots**

![image](https://user-images.githubusercontent.com/2119212/104830540-223c4280-584e-11eb-84ad-62f73125ad03.png)

**Version**

```shell
{
    "Version": "v1.9.0+e26ad30",
    "BuildDate": "2021-01-15T22:56:37Z",
    "GitCommit": "e26ad305b41f706ec23d10fca64d4466d39c7ed2",
    "GitTreeState": "clean",
    "GoVersion": "go1.14.12",
    "Compiler": "gc",
    "Platform": "linux/amd64",
    "KsonnetVersion": "v0.13.1",
    "KustomizeVersion": "v3.8.1 2020-07-16T00:58:46Z",
    "HelmVersion": "v3.4.1+gc4e7485",
    "KubectlVersion": "v1.17.8",
    "JsonnetVersion": "v0.17.0"
}
```

**Logs**

```
Paste any relevant application logs here.
```

---

p.s. The contrast of white text on orange is pretty poor. https://webaim.org/resources/contrastchecker/?fcolor=FFFFFF&bcolor=FFCC00

**Comment by reginapizza** [comment]
I was about to submit a PR for this but wanted to make sure I understood this a little better first... from looking at the code, the text for this could also be "Healthy | Synced", correct? Are there other options I'm missing? Is "unknown" here a possibility? I wanted to test out my spacing with all possibilities so that I know it won't mess any other options up; so far I have it looking like this

![degraded-outofsync](https://user-images.githubusercontent.com/50851526/106818081-cba07800-6645-11eb-82b5-14fdd4aff0dd.png)

![healthy-synced](https://user-images.githubusercontent.com/50851526/106817752-50d75d00-6645-11eb-896d-d57f56dd3a40.png)

As for the color, if it should be changed, what other color would you like? Just a more contrasted version of the same color or something else?

**Comment by jsoref** [comment]
I'm personally asking for contrast. Someone from the project could provide more project orientated guidance (sorry, I'm wearing an end user /accessibility hat 🎩 here).

You might look at what/how other groups/projects that manage badges do to when making yellow badges.

The lazy approach is black on yellow instead of white on yellow.

**Comment by reginapizza** [comment]
what if it was a color like this? I chose blue for contrast + is supposed to be a complementary color to the coral color, contrast for this is 3.01:1. 

Alternatively, purple like the helmet image on the badge itself, contrast for that one is 3.22:1

![argocd-badge](https://user-images.githubusercontent.com/50851526/106987353-3ed7e600-673b-11eb-959f-850602cda864.png)
![argocd-badge2](https://user-images.githubusercontent.com/50851526/106987355-40a1a980-673b-11eb-9572-4e6cf0c65fd4.png)

**Comment by jsoref** [comment]
I don't think you can use blue (that's generally the color used for progressing).

Out of sync's current indicator looks like this:
![image](https://user-images.githubusercontent.com/6354229/101232187-ea45bb00-3664-11eb-95a2-5000829f852e.png)
-- Picture stolen from a random issue (we don't seem to have any out of sync objects right now, I suppose I should be happy?)

One approach would be to use the icon and then stick the text into a non color-specific box.

**Comment by jsoref** [comment]
Fwiw, all other views include icons e.g. ⏸️ for paused or 👻 for missing. or ⬆️  for out of sync. I think I'd like those icons to be included in the badges. It will add some consistency between the normal ArgoCD views and the badging.

**Comment by reginapizza** [comment]
I like the idea of adding the badges and I think having the color behind only the associated icon instead of the wording would both look good and add greater contrast to better see the wording. However, this badge is already very crowded and I imagine that adding another icon would require the width of the badge to be increased. I understand that the badge can be taken and used on a user's README/website ([according to the docs](https://argoproj.github.io/argo-cd/user-guide/status-badge/)). Would changing the size possibly mess with user's websites if they're expecting a certain size for the badge and suddenly that's changed?

**Comment by jsoref** [comment]
Potentially, but most places tend to just have an area for a run of badges and hopefully wouldn't mind a lot.
